src/debug.rs

```diff
@@ -140,7 +140,10 @@ impl Config {
             .with(
                 tracing_subscriber::fmt::layer()
                     .json()
-                    .with_span_events(FmtSpan::FULL)
+                    .flatten_event(true)
+                    .with_thread_ids(true)
+                    .with_span_list(false)
+                    .with_span_events(FmtSpan::ACTIVE)
```

Comment:
- `flatten_event`[^1]
  - This is just a minor simplification of the json structure that should save a bit of space
- `with_span_list`[^2] and `with_thread_ids`[^3]
  - Idea here is that we don't need to add the span list to every trace if we include the thread id
  - We can infer the span list by looking at the previously entered/exited spans on the same thread id for a given trace
- `with_span_events`[^4]
  - Switching to `ACTIVE` as that includes all enters/exits, which seem to  me to be far more relevant for debugging than creates/drops

[^1]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.flatten_event
[^2]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_span_list
[^3]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_thread_ids
[^4]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_span_events

Reply:
> `flatten_event`

Do you have examples of what this looks like before/after?

> `with_span_list` and `with_thread_ids`
> Idea here is that we don't need to add the span list to every trace if we include the thread id

Is the idea that by recording thread IDs, we can retain the context across temporally intertwined span events without actually including them in the message?

If so, this won't actually do what you're after: `tokio` is multithreaded, yes, but only as an implementation detail; it is primarily a work-stealing executor of concurrent lightweight tasks. 

In other words, the same thread can work on many disparate tasks that have no relation to one another. So unfortunately I don't think this'll help.

Reply:
> Do you have examples of what this looks like before/after?

Sure! Before:
```json
{"timestamp":"2024-09-19T21:32:44.075971Z","level":"INFO","fields":{"message":"enter"},"target":"broker::db::sqlite","span":{"location":"\"/Users/nic/.config/fossa/broker/db.sqlite\"","name":"connect"},"threadId":"ThreadId(1)"}
```
After:
```json
{"timestamp":"2024-09-19T19:38:11.111064Z","level":"INFO","message":"enter","target":"broker::db::sqlite","span":{"location":"\"/Users/nic/.config/fossa/broker/db.sqlite\"","name":"connect"},"threadId":"ThreadId(1)"}
```

> If so, this won't actually do what you're after: tokio is multithreaded, yes, but only as an implementation detail; it is primarily a work-stealing executor of concurrent lightweight tasks.
>
> In other words, the same thread can work on many disparate tasks that have no relation to one another. So unfortunately I don't think this'll avoid that problem.

Damn, okay, good to know. I was hoping that this would be a cheeky built-in way to accomplish what you mentioned in [this thread](https://teamfossa.slack.com/archives/C039KE5ERNE/p1726691522737939?thread_ts=1726686450.086599&cid=C039KE5ERNE) but alas. I'll take another stab at it!